Doc/c-api/unicode.rst

```diff
@@ -1600,6 +1600,8 @@ object.
 
    Discard the internal Unicode buffer and destroy the writer instance.
 
+   Do nothing if *writer* is ``NULL``.
```

Comment:
From the `PyMem_Free()` documentation: "``` If *p* is ``NULL``, no operation is performed. ```"

From the `Py_CLEAR()` documentation: "``` The object may be ``NULL``, in which case the macro has no effect ```".

 Does anything of it look better?

`PyInitConfig_Free()` does not support NULL (I think that it should). Is there other explicit close/destruction API?

Reply:
> From the PyMem_Free() documentation: "If *p* is ``NULL``, no operation is performed."

I copied this doc.